File: Shared/Blog.Contracts/V1/Responses/HealthChecks/HealthCheckResponse.cs
using System;
using System.Collections.Generic;

namespace Blog.Contracts.V1.Responses.HealthChecks
{
    /// <summary>
    /// Health check response.
    /// </summary>
    public class HealthCheckResponse
    {
        /// <summary>
        /// Gets or sets the status.
        /// </summary>
        /// <value>
        /// The status.
        /// </value>
        public string Status { get; set; }

        /// <summary>
        /// Gets or sets the checks.
        /// </summary>
        /// <value>
        /// The checks.
        /// </value>
        public IEnumerable<HealthCheck> Checks { get; set; }

        /// <summary>
        /// Gets or sets the duration.
        /// </summary>
        /// <value>
        /// The duration.
        /// </value>
        public TimeSpan Duration { get; set; }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Responses/HealthChecks/HealthCheckResponse.fs
namespace Blog.FSharp.Contracts.V1.Responses.Chart.HealthChecks

open System
open System.Collections.Generic

/// <summary>
/// Health check response.
/// </summary>
type HealthCheckResponse() =
    /// <summary>
    /// Gets or sets the status.
    /// </summary>
    member val Status: string = "" with get, set

    /// <summary>
    /// Gets or sets the checks.
    /// </summary>
    member val Checks: IEnumerable<HealthCheck> = Seq.empty with get, set

    /// <summary>
    /// Gets or sets the duration.
    /// </summary>
    member val Duration: TimeSpan = TimeSpan.Zero with get, set

